query-projector/src/lib.rs

```diff
@@ -116,6 +116,29 @@ impl QueryResults {
             &FindRel(_)    => Box::new(|| QueryResults::Rel(vec![])),
         }
     }
+
+    pub fn to_pretty(&self) -> Result<String> {
+        use QueryResults::*;
+        match self {
+            &Scalar(Some(ref val)) => { 
+                Ok(val.value_type().to_edn_value().to_pretty(120).unwrap_or(String::new()) )
+            },
+            &Tuple(Some(ref vals)) => { 
+                Ok(vals.iter().map(|val| format!("{}\t", val.value_type().to_edn_value().to_pretty(120).unwrap_or(String::new()))).collect::<Vec<String>>().into_iter().collect::<String>())
+            },
+            &Coll(ref vals) => { 
+                Ok(vals.iter().map(|val| format!("{}\n", val.value_type().to_edn_value().to_pretty(120).unwrap_or(String::new()))).collect::<Vec<String>>().into_iter().collect::<String>())
+            },
+            &Rel(ref valsvec) => { 
+                let mut output = String::new();
+                for vals in valsvec {
+                    output.push_str(&format!("{}\n", &vals.iter().map(|val| format!("{}\t", val.value_type().to_edn_value().to_pretty(120).unwrap_or(String::new()))).collect::<Vec<String>>().into_iter().collect::<String>()));
```

Comment:
My point was that if you actually had EDN -- a vector of vectors for `Rel` -- then you just pretty print the whole thing!  Now, it sucks to make an owned EDN struct just for printing, but this is just for debug output, right?  This will give far more consistent output.

If you really don't want to produce EDN<, i might be better to do this using `pretty.rs` directly.

In any case, this really doesn't make sense: your tabs _will not line up_ if the leading column doesn't fit within a single tabstop.  If you want to do this, do it properly with the tab writer crate (used already in `db`).